packages/adapters/bitcoin/src/connectors/OKXConnector.ts

```diff
@@ -16,63 +17,99 @@ import { UnitsUtil } from '../utils/UnitsUtil.js'
 
 const OKX_NETWORK_KEYS = {
   [bitcoin.caipNetworkId]: 'bitcoin',
-  [bitcoinTestnet.caipNetworkId]: 'bitcoinTestnet'
+  [bitcoinTestnet.caipNetworkId]: 'bitcoinTestnet',
+  [bitcoinSignet.caipNetworkId]: 'bitcoinSignet'
 } as const
 
+declare global {
+  interface Window {
+    okxwallet: {
+      bitcoin: OKXConnector.Wallet
+      bitcoinTestnet: OKXConnector.Wallet
+      bitcoinSignet: OKXConnector.Wallet
+      cardano: { icon: string }
+    }
+  }
+}
+
 export class OKXConnector extends ProviderEventEmitter implements BitcoinConnector {
   public readonly id = 'OKX'
   public readonly name = 'OKX Wallet'
   public readonly chain = 'bip122'
   public readonly type = 'ANNOUNCED'
   public readonly explorerId =
     PresetsUtil.ConnectorExplorerIds[CommonConstantsUtil.CONNECTOR_ID.OKX]
-  public readonly imageUrl: string
   public readonly requestedCaipNetworkId?: CaipNetwork['caipNetworkId']
+  public readonly imageUrl: string
 
   public readonly provider = this
 
-  private wallet: OKXConnector.Wallet
   private readonly requestedChains: CaipNetwork[] = []
 
-  constructor({
-    wallet,
-    requestedChains,
-    imageUrl,
-    requestedCaipNetworkId
-  }: OKXConnector.ConstructorParams) {
+  constructor({ requestedChains, requestedCaipNetworkId }: OKXConnector.ConstructorParams) {
     super()
-    this.wallet = wallet
     this.requestedChains = requestedChains
-    this.imageUrl = imageUrl
     this.requestedCaipNetworkId = requestedCaipNetworkId
+    this.imageUrl = typeof window === 'undefined' ? '' : window.okxwallet?.cardano?.icon || ''
   }
 
   public get chains() {
-    return this.requestedChains.filter(
-      chain =>
-        chain.caipNetworkId ===
-        ChainController.getActiveCaipNetwork(ConstantsUtil.CHAIN.BITCOIN)?.caipNetworkId
-    )
+    return this.requestedChains
   }
 
-  public async connect(): Promise<string> {
-    const result = await this.wallet.connect()
+  public async connect(params?: { caipNetworkId?: CaipNetworkId }): Promise<string> {
+    const caipNetworkId =
+      params?.caipNetworkId ??
+      ChainController.getActiveCaipNetwork(ConstantsUtil.CHAIN.BITCOIN)?.caipNetworkId
 
-    this.bindEvents()
+    if (!caipNetworkId) {
+      throw new Error('No active network available')
+    }
 
-    this.emit('accountsChanged', [result.address])
+    const currentWallet = this.getWallet()
+    if (currentWallet) {
+      this.unbindEvents({ wallet: currentWallet })
+    }
+    const wallet = this.getWallet({ requestedCaipNetworkId: caipNetworkId })
+
+    if (!wallet) {
+      throw new Error('No wallet available')
+    }
+
+    this.bindEvents({ wallet })
+
+    const result = await wallet.connect()
 
     return result.address
   }
 
   public async disconnect(): Promise<void> {
-    this.unbindEvents()
-    await this.wallet.disconnect()
+    const wallet = this.getWallet()
+    if (!wallet) {
+      throw new Error('No wallet available')
+    }
+    this.unbindEvents({ wallet })
```

Comment:
Should this be after `await wallet.disconnect()` to make sure disconnection is done first